### PR TITLE
Small improvements for mTLS

### PR DESCRIPTION
**Pull request by krucod3**
Issues: #6

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [ ] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: tools/dev_scripts/ankaios-secure-start [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+
+echo -e "${RED}To use certificates for the ank cli run the cli as follows:${NC}"
+echo 'ank --ca_pem /workspaces/ankaios/.certs/ca.pem --crt_pem /workspaces/ankaios/.certs/cli.pem --key_pem /workspaces/ankaios/.certs/cli-key.pem get workloads'
+
+echo ''
+
+echo -e "${RED}or export the following variables:${NC}"
+echo unset ANK_INSECURE \
```
Comment (krucod3) [inline tools/dev_scripts/ankaios-secure-start:34]: We don't need this

File: tools/dev_scripts/ankaios-secure-start [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export ANK_CA_PEM="/workspaces/ankaios/$CERTS_FOLDER_NAME/ca.pem" \
```
Comment (krucod3) [inline tools/dev_scripts/ankaios-secure-start:35]:
```suggestion
echo export ANK_CA_PEM="/workspaces/ankaios/$CERTS_FOLDER_NAME/ca.pem" \
```

File: tools/dev_scripts/ankaios-secure-start [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ANKAGENT_CA_PEM="$CERTS_DIR/ca.pem" \
+ANKAGENT_CRT_PEM="$CERTS_DIR/agent.pem" \
+ANKAGENT_KEY_PEM="$CERTS_DIR/agent-key.pem" \
+RUST_LOG=debug /workspaces/ankaios/target/x86_64-unknown-linux-musl/debug/ank-agent --name agent_A 2> "$LOG_DIR/ank-agent" &
+
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+
+echo -e "${RED}To use certificates for the ank cli run the cli as follows:${NC}"
+echo 'ank --ca_pem /workspaces/ankaios/.certs/ca.pem --crt_pem /workspaces/ankaios/.certs/cli.pem --key_pem /workspaces/ankaios/.certs/cli-key.pem get workloads'
```
Comment (christoph-hamm) [inline tools/dev_scripts/ankaios-secure-start:29]: Use `$CERTS_DIR` here.
```suggestion
echo "ank --ca_pem \"$CERTS_DIR/ca.pem\" --crt_pem \"$CERTS_DIR/cli.pem\" --key_pem \"$CERTS_DIR/cli-key.pem\" get workloads"
```

File: server/src/main.rs [hunk 4]
```diff
@@ -64,26 +64,14 @@ async fn main() {
...
-            // [impl->swdd~server-fails-on-missing-file-paths-and-insecure-cli-arguments~1]
-            (_, ca_pem, crt_pem, key_pem) => Err(format!(
-                "Provide the files via ANKSERVER_CA_PEM={} ANKSERVER_CRT_PEM={} ANKSERVER_KEY_PEM={} or deactivate mTLS with '-k' or '--insecure' option!",
-                ca_pem.unwrap_or(String::from("\"\"")),
-                crt_pem.unwrap_or(String::from("\"\"")),
-                key_pem.unwrap_or(String::from("\"\""))
-            )),
-        };
+    if TLSConfig::is_config_conflicting(args.insecure, &args.ca_pem, &args.crt_pem, &args.key_pem) {
+        log::warn!("Insecure and secure flags specified at the same time. Defaulting to secure communication.");
```
[the remark this answers is not in the document]
Comment (krucod3) [reply server/src/main.rs:68]: Good idea

File: grpc/src/lib.rs [hunk 5]
```diff
@@ -36,6 +36,45 @@ pub mod security {
...
+                }
+                // [impl->swdd~cli-establishes-insecure-communication-based-on-provided-insecure-cli-argument~1]
+                (true, None, None, None) => Ok(None),
+                // [impl->swdd~cli-fails-on-missing-file-paths-and-insecure-cli-arguments~1]
+                (_, ca_pem, crt_pem, key_pem) => Err(format!(
+                    "Provide mTLS config via ANKAGENT_CA_PEM={} ANKAGENT_CRT_PEM={} ANKAGENT_KEY_PEM={} or deactivate mTLS with '-k' or '--insecure' option only!",
+                    ca_pem.unwrap_or(String::from("\"\"")),
+                    crt_pem.unwrap_or(String::from("\"\"")),
+                    key_pem.unwrap_or(String::from("\"\""))
+                )),
```
[the remark this answers is not in the document]
Comment (krucod3) [reply grpc/src/lib.rs:73]: Good catch 👍

File: tools/dev_scripts/ankaios-secure-start (hunk 3 above)
Comment (krucod3) [reply tools/dev_scripts/ankaios-secure-start:29]: This was done explicitly, as I didn't want to print out the CERTS_DIR as `CERTS_DIR="$SCRIPT_DIR/../../$CERTS_FOLDER_NAME"` with some `/../../` in the middle ... 
For a dev script, I think the current behavior is fine.

File: tools/dev_scripts/ankaios-secure-start [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+echo ''
+
+echo -e "${RED}or export the following variables:${NC}"
+echo export ANK_CA_PEM="/workspaces/ankaios/$CERTS_FOLDER_NAME/ca.pem" \
+export ANK_CRT_PEM="/workspaces/ankaios/$CERTS_FOLDER_NAME/cli.pem" \
+export ANK_KEY_PEM="/workspaces/ankaios/$CERTS_FOLDER_NAME/cli-key.pem"
```
[the remark this answers is not in the document]
Comment (krucod3) [reply tools/dev_scripts/ankaios-secure-start:36]: I don't like the `$CERTS_DIR`, see above.

**Review by christoph-hamm — APPROVED** [review]

**Pull request merged**